### PR TITLE
[Mobile 1531] - Add date attributes support to React Native

### PR DESCRIPTION
**Pull request by khmMouna**
### What do these changes do?
Add date attributes support to React Native

### How did you verify these changes?
Using the sample App.

PS: Need a rebase after merging the [268 PR](https://github.com/urbanairship/react-native-module/pull/268 ) to fix the github-actions CI / ios job. 

Update: 
- [x] Rebase done and checks passed

### CODE REVIEW

File: src/android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java [hunk 1]
```diff
@@ -914,23 +911,20 @@ private static void applyAttributeOperations(@NonNull AttributeEditor editor, @N
...
-                    } catch (ParseException e) {
-                        editor.setAttribute(key, value);
-                    }
-                } else if (ReadableType.Number == type) {
+                    editor.setAttribute(key, value);
+                } else if ("number".equals(valueType)) {
                     double value = operation.getDouble(ATTRIBUTE_OPERATION_VALUE);
                     editor.setAttribute(key, value);
+                } else if ("date".equals(valueType)) {
+                    double value = operation.getDouble(ATTRIBUTE_OPERATION_VALUE);
```
[the remark this answers is not in the document]
Comment (khmMouna) [reply src/android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java:924]: **getLong** does not exist.

**Comment by khmMouna** [comment]
@BrianBatchelder  do we need to support date for named user attributes ? If yes can I do it with this ticket or should I create another ticket for that?

File: src/ios/UARCTModule/UrbanAirshipReactModule.m [hunk 2]
```diff
@@ -555,22 +555,29 @@ - (RCTBridge *)bridge {
 
 - (UAAttributeMutations *)mutationsWithOperations:(NSArray *)operations {
     UAAttributeMutations *mutations = [UAAttributeMutations mutations];
-
     for (NSDictionary *operation in operations) {
```
Comment (khmMouna) [inline src/ios/UARCTModule/UrbanAirshipReactModule.m:558]:
```suggestion

    for (NSDictionary *operation in operations) {
```

File: src/ios/UARCTModule/UrbanAirshipReactModule.m [hunk 3]
```diff
@@ -562,15 +562,23 @@ - (UAAttributeMutations *)mutationsWithOperations:(NSArray *)operations {
...
+                    [mutations setDate:date forAttribute:name];
+                } else {
+                    UA_LWARN("Unknown channel attribute type: %@", valueType);
+                }
         } else if ([action isEqualToString:@"remove"]) {
-            [mutations removeAttribute:name];
+                [mutations removeAttribute:name];
         }
     }
+
```
Comment (khmMouna) [inline src/ios/UARCTModule/UrbanAirshipReactModule.m:581]:
```suggestion
```

File: src/ios/UARCTModule/UrbanAirshipReactModule.m [hunk 4]
```diff
@@ -562,13 +562,20 @@ - (UAAttributeMutations *)mutationsWithOperations:(NSArray *)operations {
...
+                } else if ([valueType isEqualToString:@"date"]) {
+                    // JavaScript's date type doesn't pass through the JS to native bridge. Dates are instead serialized as milliseconds since epoch.
+                    NSDate *date = [NSDate dateWithTimeIntervalSince1970:[(NSNumber *)value doubleValue] / 1000.0];
+                    [mutations setDate:date forAttribute:name];
+                } else {
+                    UA_LWARN("Unknown channel attribute type: %@", valueType);
+                }
         } else if ([action isEqualToString:@"remove"]) {
-            [mutations removeAttribute:name];
+                [mutations removeAttribute:name];
```
Comment (khmMouna) [inline src/ios/UARCTModule/UrbanAirshipReactModule.m:578]:
```suggestion
            [mutations removeAttribute:name];
```

File: src/android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (BrianBatchelder) [reply src/android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java:938]: Shoot. I'm concerned that maybe `((Number) value).longValue()` doesn't work in this case. Can `double` be casted to `Number`? You might want to use `Double.valueOf(value).longValue()`.

File: src/js/AttributeEditor.js [hunk 5]
```diff
@@ -23,8 +23,23 @@ class AttributeEditor {
    * @param {string} name The attribute name.
    * @return {AttributeEditor} The attribute editor instance.
    */
-  setAttribute(name: string, value: string|number): AttributeEditor {
+  setAttribute(name: string, value: string|number|date): AttributeEditor {
```
Comment (BrianBatchelder) [inline src/js/AttributeEditor.js:26]:
```suggestion
  setAttribute(name: string, value: string|number|Date): AttributeEditor {
```

File: src/js/AttributeEditor.js [hunk 6]
```diff
@@ -23,8 +23,23 @@ class AttributeEditor {
    * @param {string} name The attribute name.
```
Comment (BrianBatchelder) [inline src/js/AttributeEditor.js:23]:
```suggestion
   * @param {string|number|Date} name The attribute name.
```

File: src/android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (khmMouna) [reply src/android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java:938]: The three solutions work as expected. I choose the simplest one :D `(long)value`

**Pull request merged**